pkg/webui/components/dropdown-v2/index.js

```diff
@@ -36,12 +36,13 @@ const Dropdown = ({ className, children, larger, onItemsClick }) => {
         const parentRect = ref.current.parentElement.getBoundingClientRect()
         const spaceBelow = window.innerHeight - parentRect.bottom
         const spaceAbove = parentRect.top
-        const spaceLeft = parentRect.left
-        const spaceRight = window.innerWidth - parentRect.right
+        const dropdownHeight = ref.current.clientHeight
+        const dropdownWidth = ref.current.clientWidth
+        const spaceOnLeft = parentRect.left
 
-        setIsBelow(spaceBelow > ref.current.clientHeight || spaceAbove < ref.current.clientHeight)
+        setIsBelow(spaceBelow > dropdownHeight || spaceAbove < dropdownHeight)
 
-        setIsOnRight(spaceRight < spaceLeft)
+        setIsOnRight(spaceOnLeft > dropdownWidth)
       }
     }
```

Comment:
The problem with this logic is that it is only executed after the first render, which sometimes leads to layout flashes. To ways around this:
1. Render the dropdown invisible initially `visibility: hidden` and remove that once the position is determined.
2. Calculate the height deterministically based on the number of items and then set the position before rendering. This should work with simple dropdowns where the height of the items stays consistent.

Reply:
I have added event listeners on scroll and resize in case there was a screen changes while the dropdown is open. Let me know if this is what you meant how the dropdown menu should behave.

https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/943fb75a-0d7a-41d0-92c1-1494a854ce3a

Reply:
Good to have these but that was not what my comment was about. What I meant was layout flashes when the dropdown first opens.

https://github.com/TheThingsNetwork/lorawan-stack/assets/5710611/0aa4d9f2-dc0a-43f0-a8a4-11eebf79dba5

It doesn't happen always but the video shows how the dropdown sometimes flashes up above the button briefly. This is on chrome, but I'm pretty sure this happens on other browsers as well.

I believe this can be addressed in two ways:

> 1. Render the dropdown invisible initially `visibility: hidden` and remove that once the position is determined.
> 2. Calculate the height deterministically based on the number of items and then set the position before rendering. This should work with simple dropdowns where the height of the items stays consistent.